storage/kv.go

```diff
@@ -103,3 +103,15 @@ type WatchableKV interface {
 	// should always call cancel as soon as watch is done.
 	Watcher(key []byte, prefix bool, startRev, endRev int64) (Watcher, CancelFunc)
 }
+
+// ConsistentWatchableKV is a WatchableKV that maintains index of the latest txn.
+// It guarantees that the maintained index is consistent with the latest txn,
+// which means that the index always indicates the latest txn happened in
+// ConsistentWatchableKV in all scenarios, even if program crashes at any time.
+type ConsistentWatchableKV interface {
+	WatchableKV
+
+	// Index returns the index of the latest txn.
+	// It is blocked if there is on-going txn.
+	Index() uint64
```

Comment:
Good. So why it is index of the latest txn? Shouldnt it be the index of the last consistent entry? The whole point is that consistentStore knows about the consistent algo index. You missed this point in this pull request. We do not need a txn centric concept. We need a consistent log centric concept.